src/dsp_tools/commands/xmlupload/resource_create_client.py

```diff
@@ -351,14 +362,26 @@ def _make_time_value(value: XMLValue) -> dict[str, Any]:
     }
 
 
-def _make_uri_value(value: XMLValue) -> dict[str, Any]:
-    return {
-        "@type": "knora-api:UriValue",
-        "knora-api:uriValueAsUri": {
-            "@type": "xsd:anyURI",
-            "@value": value.value,
-        },
-    }
+def _serialise_uri_prop(prop: XMLProperty, permissions_lookup: dict[str, Permissions]) -> dict[str, Any]:
+    uri_values: list[SerialiseValue] = []
+    for v in prop.values:
+        value_permission = cast(str, v.permissions)
```

Comment:
why is this cast needed?

Reply:
Because in the class XMLValue permissions are optional, then it does not accept me to retrieve it from the dict. This is a problem that would be solved when we separate the classes.